nova/virt/vmwareapi/vmops.py

```diff
@@ -2014,10 +2017,44 @@ def finish_migration(self, context, migration, instance, disk_info,
         """Completes a resize, turning on the migrated instance."""
         vm_ref = vm_util.get_vm_ref(self._session, instance)
 
-        # 5. Update ephemerals if necessary
+        flavor = instance.flavor
+        boot_from_volume = compute_utils.is_volume_backed_instance(context,
+                                                                   instance)
+        reattach_volumes = False
+        # 2. Relocate the VM if necessary
+        if self._cluster_name != get_cluster_name_from_dest_host(
+                migration.dest_host):
```

Comment:
Does that `if` work? Isn't `dest_host` the one we're currently running on? The `self._relocate_vm()` also seems to relocate to the current host (it uses `self._root_resource_pool`), so shouldn't this `if` rather check if the destination cluster is our cluster instead of `!=`?